File: 3605-construct-the-minimum-bitwise-array-i/construct-the-minimum-bitwise-array-i.py
from typing import List

class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        ans = []
        for n in nums:
            if n == 2:
                ans.append(-1)
            else:
                # Find the smallest x such that x | (x + 1) == n
                # Since constraints are small (<= 1000), we can loop
                found = False
                for x in range(n):
                    if (x | (x + 1)) == n:
                        ans.append(x)
                        found = True
                        break
                
                # If no x is found (though logic suggests checking n=2 handles most),
                # append -1 just in case.
                if not found:
                    ans.append(-1)
        return ans

File: 3605-construct-the-minimum-bitwise-array-i/construct-the-minimum-bitwise-array-i.java
class Solution {
    public int[] minBitwiseArray(List<Integer> nums) {
        int n = nums.size();
        int[] ans = new int[n];

        for (int i = 0; i < n; i++) {
            int target = nums.get(i);
            int res = -1;

            // Special case for 2 (the only even prime)
            if (target == 2) {
                ans[i] = -1;
                continue;
            }

            // Iterate from 0 to target to find the smallest x
            for (int x = 0; x < target; x++) {
                if ((x | (x + 1)) == target) {
                    res = x;
                    break; 
                }
            }
            ans[i] = res;
        }
        return ans;
    }
}